File: lib/csharp/se/lth/control/labcomm2014/WrappingDecoder.cs
using System;

namespace se.lth.control.labcomm2014 {

  public class WrappingDecoder: Decoder {

    private Decoder decoder;

    public WrappingDecoder(Decoder decoder) {
      this.decoder = decoder;
    }

    public virtual void runOne() {
      decoder.runOne();
    }

    public virtual void run() {
      decoder.run();
    }    

    public virtual void register(SampleDispatcher dispatcher, 
		  SampleHandler handler) {
      decoder.register(dispatcher, handler);
    }
                  
    public virtual void registerSampleRef(SampleDispatcher dispatcher) {
      decoder.registerSampleRef(dispatcher);
    }

    public virtual bool decodeBoolean() {
      return decoder.decodeBoolean();
    }
    
    public virtual byte decodeByte() {
      return decoder.decodeByte();
    }
    
    public virtual short decodeShort() {
      return decoder.decodeShort();
    }
    
    public virtual int decodeInt() {
      return decoder.decodeInt();
    }
    
    public virtual long decodeLong() {
      return decoder.decodeLong();
    }
    
    public virtual float decodeFloat() {
      return decoder.decodeFloat();
    }
    
    public virtual double decodeDouble() {
      return decoder.decodeDouble();
    }
    
    public virtual String decodeString() {
      return decoder.decodeString();
    }
    
    public virtual int decodePacked32() {
      return decoder.decodePacked32();
    }
    
    public virtual SampleDispatcher decodeSampleRef() {
      return decoder.decodeSampleRef();
    }
    
  }

} 

File: lib/java/se/lth/control/labcomm2014/WrappingDecoder.java
package se.lth.control.labcomm2014;

import java.io.IOException;

public class WrappingDecoder implements Decoder {

  private Decoder decoder;

  public WrappingDecoder(Decoder decoder) {
    this.decoder = decoder;
  }

  public void runOne() throws Exception {
    decoder.runOne();
  }
  
  public void run() throws Exception {
    decoder.run();
  }    
  
  public void register(SampleDispatcher dispatcher, 
                       SampleHandler handler) throws IOException{
    decoder.register(dispatcher, handler);
  }
  
  public void registerSampleRef(SampleDispatcher dispatcher) throws IOException {
    decoder.registerSampleRef(dispatcher);
  }
  
  public boolean decodeBoolean() throws IOException {
    return decoder.decodeBoolean();
  }
  
  public byte decodeByte() throws IOException {
    return decoder.decodeByte();
  }
  
  public short decodeShort() throws IOException {
    return decoder.decodeShort();
  }
  
  public int decodeInt() throws IOException {
    return decoder.decodeInt();
  }
  
  public long decodeLong() throws IOException {
    return decoder.decodeLong();
  }
  
  public float decodeFloat() throws IOException {
    return decoder.decodeFloat();
  }
  
  public double decodeDouble() throws IOException {
    return decoder.decodeDouble();
  }
  
  public String decodeString() throws IOException {
    return decoder.decodeString();
  }
  
  public int decodePacked32() throws IOException {
    return decoder.decodePacked32();
  }
  
  public SampleDispatcher decodeSampleRef() throws IOException {
    return decoder.decodeSampleRef();
  }
  
}
